### PR TITLE
Add pull request template

### PR DESCRIPTION
**Pull request by mvdbeek**
I have recently spent a good amount of time posting various aspects of this on PRs, I hope this will make the whole process a little smoother.

This is what it looks like when you open a PR:

Screenshot 2024-11-26 at 10 30 42

### CODE REVIEW

**Review by bernt-matthias — APPROVED** [review]

**Review by lldelisle — APPROVED** [review]
Good

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+* [ ] In workflow: workflow inputs and outputs have human readable names (spaces are fine, no underscore, dash only where spelling dictates it), no abbreviation unless it is generally understood. Altering input or output labels requires adjusting these labels in the the workflow-tests.yml file as well
+* [ ] In workflow: `name` field should be human readable (spaces are fine, no underscore, dash only where spelling dictates it), no abbreviation unless generally understood
+* [ ] Workflow folder: prefer dash (`-`) over underscore (`_`), prefer all lowercase. Folder becomes repository in [iwc-workflows organization](https://github.com/iwc-workflows) and is included in TRS id
+* [ ] Readme explains what workflow does, what are valid inputs and what outputs users can expect. If a tutorial or other resources exist they can be linked. If a similar workflow exists in IWC readme should explain differences with existing workflow and when one might prefer one workflow over another
+* [ ] Changelog contains appropriate entries
+* [ ] Large files (> 100 KB) are uploaded to zenodo and location urls are used in test file
```
Comment (mvdbeek) [inline .github/PULL_REQUEST_TEMPLATE.md:14]:
```suggestion
* [ ] Large files (> 100 KB) are uploaded to zenodo and location urls are used in test file
* [ ] No email post job action
```
?

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,14 @@
+FOR CONTRIBUTOR:
+* [ ] I have read the [Adding workflows guidelines](https://github.com/galaxyproject/iwc/blob/main/workflows/README.md#adding-workflows)
+* [ ] License permits unrestricted use (educational + commercial)
```
Comment (mvdbeek) [inline .github/PULL_REQUEST_TEMPLATE.md:3]: We could put that into the IWC profile for the linter

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+* [ ] In workflow: workflow inputs and outputs have human readable names (spaces are fine, no underscore, dash only where spelling dictates it), no abbreviation unless it is generally understood. Altering input or output labels requires adjusting these labels in the the workflow-tests.yml file as well
+* [ ] In workflow: `name` field should be human readable (spaces are fine, no underscore, dash only where spelling dictates it), no abbreviation unless generally understood
+* [ ] Workflow folder: prefer dash (`-`) over underscore (`_`), prefer all lowercase. Folder becomes repository in [iwc-workflows organization](https://github.com/iwc-workflows) and is included in TRS id
+* [ ] Readme explains what workflow does, what are valid inputs and what outputs users can expect. If a tutorial or other resources exist they can be linked. If a similar workflow exists in IWC readme should explain differences with existing workflow and when one might prefer one workflow over another
+* [ ] Changelog contains appropriate entries
```
Comment (mvdbeek) [inline .github/PULL_REQUEST_TEMPLATE.md:13]: @lldelisle mentioned that we might need to improve the linting in planemo

**Pull request merged**